parlai/core/torch_generator_agent.py

```diff
@@ -752,6 +758,7 @@ def compute_loss(self, batch, return_output=False):
         self.record_local_metric(
             'token_em', AverageMetric.many(num_tokens_correct == num_target_tokens)
         )
+        self.custom_loss_metrics(batch, loss_per_token)
```

Comment:
nit: could we rename to `record_per_token_metrics` or something similar? I want to make sure people understand that this is function is specifically provided the `loss_per_token`